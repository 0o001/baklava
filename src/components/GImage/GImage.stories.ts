// @ts-nocheck
import { storiesOf } from '@storybook/vue';
import { withKnobs, text } from '@storybook/addon-knobs';
import GImage from './GImage.vue';

const stories = storiesOf('GImage', module);
stories.addDecorator(withKnobs);

stories
  .addParameters({ component: GImage })
  .add('default', () => ({
    components: { GImage },
    template: '<GImage />',
  }))
  .lokiSkip('src', () => ({
    components: { GImage },
    props: {
      src: {
        default: text('Source', 'https://picsum.photos/500?random'),
      },
      width: {
        default: text('Width', ''),
      },
      height: {
        default: text('Height', ''),
      },
      maxWidth: {
        default: text('Max Width', '500'),
      },
      maxHeight: {
        default: text('Max Height', '500'),
      },
    },
    template: `<GImage
                :src="src"
                :width="width"
                :height="height"
                :maxWidth="maxWidth"
                :maxHeight="maxHeight" />`,
  }))
  .lokiSkip('src lazy load', () => ({
    components: { GImage },
    props: {
      src: {
        default: text('Source', 'https://picsum.photos/500?random'),
      },
      defaultImage: {
        default: text('Default Image', 'https://cdn.dsmcdn.com/seller-center/spm/seller-center-product/assets/default.jpg'),
      },
      width: {
        default: text('Width', ''),
      },
      height: {
        default: text('Height', ''),
      },
      maxWidth: {
        default: text('Max Width', '500'),
      },
      maxHeight: {
        default: text('Max Height', '500'),
      },
    },
    template: `
    <div>
          <GImage
                :src="src"
                :defaultImage="defaultImage"
                :width="width"
                :height="height"
                :maxWidth="maxWidth"
                :maxHeight="maxHeight" />
         <GImage
                :src="src"
                :defaultImage="defaultImage"
                :width="width"
                :height="height"
                :maxWidth="maxWidth"
                :maxHeight="maxHeight" />
          <GImage
                :src="src"
                :defaultImage="defaultImage"
                :width="width"
                :height="height"
                :maxWidth="maxWidth"
                :maxHeight="maxHeight" />
          <GImage
                :src="src"
                :defaultImage="defaultImage"
                :width="width"
                :height="height"
                :maxWidth="maxWidth"
                :maxHeight="maxHeight" />
          <GImage
                :src="src"
                :defaultImage="defaultImage"
                :width="width"
                :height="height"
                :maxWidth="maxWidth"
                :maxHeight="maxHeight" />
          <GImage
                :src="src"
                :defaultImage="defaultImage"
                :width="width"
                :height="height"
                :maxWidth="maxWidth"
                :maxHeight="maxHeight" />
          <GImage
                :src="src"
                :defaultImage="defaultImage"
                :width="width"
                :height="height"
                :maxWidth="maxWidth"
                :maxHeight="maxHeight" />
    </div>`,
  }));
