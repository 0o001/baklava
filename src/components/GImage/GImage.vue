<template>
  <div class="g-image">
    <img
      lazy
      :src="isSrcUrl ? src : require(src)"
      :width="width"
      :maxWidth="maxWidth"
      :height="height"
      :maxHeight="maxHeight"
    >
  </div>
</template>

<script lang="ts">

import { isUrl } from '../../utils/regex.ts';

export default {
  name: 'GImage',
  props: {
    src: {
      type: String,
      default: 'https://cdn.dsmcdn.com/shared/images/trendyol.jpeg',
    },
    width: {
      type: [String, Number],
      default: null,
    },
    height: {
      type: [String, Number],
      default: null,
    },
    maxWidth: {
      type: [String, Number],
      default: null,
    },
    maxHeight: {
      type: [String, Number],
      default: null,
    },
  },
  computed: {
    isSrcUrl () {
      return isUrl(this.src);
    },
  },
};
</script>

<style lang="scss" scoped>
  .g-image {
    display: flex;
    align-items: center;
    overflow: hidden;
  }
</style>
