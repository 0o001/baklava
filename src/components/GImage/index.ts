import GImage from './GImage.vue';

export default GImage;
