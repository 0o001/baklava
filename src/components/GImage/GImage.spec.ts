import { shallowMount } from '@vue/test-utils';
import GImage from './';

describe('GImage', () => {
  let wrapper: any;

  beforeEach(() => {
    const mockIntersectionObserver = jest.fn();
    mockIntersectionObserver.mockReturnValue({
      observe: () => null,
      unobserve: () => null,
      disconnect: () => null,
    });
    window.IntersectionObserver = mockIntersectionObserver;
    wrapper = shallowMount(GImage);
  });

  it('should match snapshot when show is prop is given as false ', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render image as default src url when src prop does not set', () => {
    expect(wrapper.vm.src).toEqual('https://cdn.dsmcdn.com/shared/images/trendyol.jpeg');
  });

  it('should render image with given src url', async () => {
    // given
    wrapper.setProps({
      src: 'https://picsum.photos/500?random',
    });

    // when
    await wrapper.vm.$nextTick();

    // then
    expect(wrapper.vm.isSrcUrl).toBeTruthy();
    expect(wrapper.vm.src).toEqual('https://picsum.photos/500?random');
  });

  it('should render image with given src absolute path', async () => {
    // given
    wrapper.setProps({
      src: '../../resources/default.jpg',
    });

    // when
    await wrapper.vm.$nextTick();

    // then
    expect(wrapper.vm.isSrcUrl).toBeFalsy();
    expect(wrapper.vm.src).toEqual('../../resources/default.jpg');
  });
});
