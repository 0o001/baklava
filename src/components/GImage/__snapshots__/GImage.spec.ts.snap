// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GImage should match snapshot when show is prop is given as false  1`] = `
<div
  class="g-image"
>
  <img
    lazy=""
    src="https://cdn.dsmcdn.com/shared/images/trendyol.jpeg"
  />
</div>
`;
